### PR TITLE
Garbage collect app change configmap in delete command

### LINKED ISSUE #644: Failure in `kapp delete` results in creation of app-change ConfigMaps with no limit on the number of app-changes
**What steps did you take:**
- Deploy an app
- Delete the app (with no permission to delete ConfigMap, to simulate a failure)

Notice that the resources are deleted but the app metadata ConfigMap still exists and there is an additional app-change ConfigMap for the last change.

- Try to delete the app again 

Notice that there is another app-change CM present now.

**What happened:**
`kapp delete` creates app change ConfigMaps which are then deleted, but in case the app deletion fails for some reason, these ConfigMaps remain on the cluster. If there is a controller or a script which tries to delete the app at certain intervals then the number of CMs goes onto increase without any limit.

**What did you expect:**
- App change ConfigMaps should not be created for delete command in the first place (this is debatable)
- In case we are creating the CMs for delete command, there should be garbage collection happening in delete just like deploy command.

**Environment:**
- kapp version (use `kapp --version`): 0.54.0
- OS (e.g. from `/etc/os-release`): darwin
- Kubernetes version (use `kubectl version`) 1.25.0

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

### PR DESCRIPTION
**Pull request by sethiyash**
#### Which issue(s) this PR fixes:
Fixes https://github.com/carvel-dev/kapp/issues/644

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional Notes for your reviewer:

| Scenarios | Garbage Collection  | App Changes Stored |
|-----------|---------------------|---------------|
| Fully Delete |  No                  |    -  |
| Partial Delete | Yes | 200
| Unsucessful Delete | Yes | 5

##### Review Checklist:

- [ ] Follows the [developer guidelines](https://carvel.dev/shared/docs/latest/development_guidelines/)
- [ ] Relevant tests are added or updated
- [ ] Relevant docs in this repo added or updated
- [ ] Relevant carvel.dev docs added or updated in a separate PR and there's
  a link to that PR
- [ ] Code is at least as readable and maintainable as it was before this
  change

#### Additional documentation e.g., Proposal, usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by rohitagg2020** [comment]
LGTM

**Comment by kumaritanushree** [comment]
LGTM

File: pkg/kapp/cmd/app/delete.go [hunk 1]
```diff
@@ -149,11 +149,26 @@ func (o *DeleteOptions) Run() error {
 		return err
 	}
 
+	if !shouldFullyDeleteApp {
+		defer func() {
+			_, numDeleted, _ := app.GCChanges(ctlapp.AppChangesMaxToKeepDefault, nil)
```
Comment (100mik) [inline pkg/kapp/cmd/app/delete.go:154]: We should probably revisit this default (200 right now I believe) at some point 🤔

**Review by 100mik — APPROVED** [review]
LGTM!

**Review by rohitagg2020 — APPROVED** [review]

**Review by praveenrewar — APPROVED** [review]

**Pull request merged**